### PR TITLE
[Don't Merge yet] Spanish Translation in progress

### PR DESCRIPTION
**Pull request by dossorio**
### Description
Spanish translation of [google-interview-university](https://github.com/jwasham/google-interview-university)

This is a work in progress ! Feel free to collaborate! 
1. Pick an issue from https://github.com/dossorio/google-interview-university/issues
2. Fork https://github.com/dossorio/google-interview-university and translate your chunk in `translations/README-es.md`
3. Open a pr with `master` of this [repo](https://github.com/dossorio/google-interview-university) as base for your pr and contribute ! Thank you so much !! :)

All commits will be squashed into one big `Spanish Translation` before merging, so don't worry about them! just translate ! 👍

### CODE REVIEW

File: README-es.md [hunk 1]
```diff
@@ -2,42 +2,44 @@
 
 Version original: [Inglés](README.md)
 
-## Qué es?
+## ¿Qué es?
 
-Es mi guía de estudio en varios meses para ir de desarrollador web (Autodidacta, sin grado en Ciencias de Computadores) a ingeniero de software en Google. 
+Es mi guía de estudio en varios meses para ir de desarrollador web (Autodidacta, sin grado en Ciencias de Computadores)
```
Comment (jameskolce) [inline README-es.md]: Creo que "Ciencias de la computación" es el termino estandar en lugar de "Ciencias de Computadoras"

**Comment by jameskolce** [comment]
I'm going to add some comments about the current translation, just little suggestions!

File: README-es.md [hunk 2]
```diff
@@ -2,42 +2,44 @@
...
+Es mi guía de estudio en varios meses para ir de desarrollador web (Autodidacta, sin grado en Ciencias de Computadores) 
+a ingeniero de software en Google. 
 
 ![Coding at the whiteboard - from HBO's Silicon Valley](https://dng5l3qzreal6.cloudfront.net/2016/Aug/coding_board_small-1470866369118.jpg)
 
-This long list has been extracted and expanded from **Google's coaching notes**, so these are the things you need to know.
-There are extra items I added at the bottom that may come up in the interview or be helpful in solving a problem. Many items are from
-Steve Yegge's "[Get that job at Google](http://steve-yegge.blogspot.com/2008/03/get-that-job-at-google.html)" and are reflected
-sometimes word-for-word in Google's coaching notes.
+Esta lista tan larga, ha sido extraída y ampliada de las **Google's coaching notes**, asi que estas son las cosas que
```
Comment (jameskolce) [inline README-es.md]: "Esta lista tan larga" → "Esta larga lista"

**Comment by jwasham** [comment]
Please move your translation file to the translations directory. Thank you!

File: README-es.md [hunk 3]
```diff
@@ -225,68 +226,70 @@ I'm on the journey, too. Follow along:
...
-Some videos are available only by enrolling in a Coursera, EdX, or Lynda.com class. These are called MOOCs.
-Sometimes the classes are not in session so you have to wait a couple of months, so you have no access. Lynda.com courses are not free.
+Algunos vídeos estan disponibles sólo al inscribirte en Coursera, EdX o Lynda.com. Éstos se llaman MOOCs (Massive Open Online Courses, 
+Cursos en línea abiertos masivos).
+A veces las clases no están disponibles, por lo que tienes que esperar un par de meses, lo que significa que no tienes acceso. 
+Los cursos de Lynda.com no son gratis.
 
-    I'd appreciate your help to add free and always-available public sources, such as YouTube videos to accompany the online course videos.
-    I like using university lectures.
+    Agradecería tu ayuda para añadir recursos públicos gratis, que siempre estuvieran disponibles, como video s de YouTube
```
Comment (MiguelMB) [inline README-es.md]: Agradecería tu ayuda para añadir recursos públicos gratis, que siempre estuvieran disponibles, como videos de YouTube

File: README-es.md [hunk 4]
```diff
@@ -225,68 +226,70 @@ I'm on the journey, too. Follow along:
...
-        - [ ] [Cracking the Coding Interview with Author Gayle Laakmann McDowell (video)](https://www.youtube.com/watch?v=aClxtDcdpsQ)
-    - [ ] How to Get a Job at the Big 4:
-        - [ ] ['How to Get a Job at the Big 4 - Amazon, Facebook, Google & Microsoft' (video)](https://www.youtube.com/watch?v=YJZCUhxNCv8)
-    - [ ] [Failing at Google Interviews](http://alexbowe.com/failing-at-google-interviews/)
+        - [ ] [Cracking the Coding Interview con el autor Gayle Laakmann McDowell (video)](https://www.youtube.com/watch?v=aClxtDcdpsQ)
+    - [ ] Como conseguir un trabajo en una de las 4 grandes:
+        - [ ] ['Como conseguir un trabajo en una de las 4 grandes - Amazon, Facebook, Google & Microsoft' (video)](https://www.youtube.com/watch?v=YJZCUhxNCv8)
+    - [ ] [Haciendo las entrevistas de Google mal](http://alexbowe.com/failing-at-google-interviews/)
 
 ## Pick One Language for the Interview
```
Comment (MiguelMB) [inline README-es.md]: ## Escoge un lenguaje de programacion para la entrevista

File: README-es.md (hunk 3 above)
Comment (dossorio) [inline README-es.md]: Good spot !! Thanks !!

Comment (dossorio) [inline README-es.md]: Done here https://github.com/dossorio/google-interview-university/commit/a749ece79534ffcec2e1ac56177d196da31bb7df @MiguelMB ! Thanks very much !

Feel free to pick any issues and start translating ! 👍 

Cheers !

**Comment by maasencioh** [comment]
Hi @dossorio, what is the state of this and how can I help?

**Comment by dossorio** [comment]
Hi @maasencioh  !

Thanks for your help !! 
Feel free to pick any of these [issues](https://github.com/dossorio/google-interview-university/issues) and open a pr !

Cheers man !

**Comment by jwasham** [comment]
This has been hanging out a while. Let me know when you want to pull @dossorio

**Comment by YoSaucedo** [comment]
I see that there is a conflict in the version of @dossorio branch with the actual repo.
I made a fork of the actual and do some translations that are missed. 

What could be the best practice? 

Do a pull request to @dossorio branch or do another pull request to this repository?

**Comment by YoSaucedo** [comment]
Trying to do a useful translation I did [a propose](https://github.com/YoSaucedo/coding-interview-university/commit/053574ffb5fe6698eb0359fbce2c736b5def39c0) to the structure of translations.
Please @jwasham let me know if this can be merged when I finish it.

**Comment by dossorio** [comment]
Hi @YoSaucedo thanks for your help !
The process we were following is the description on this pr.

Feel free to merge your pr in https://github.com/dossorio/google-interview-university/ ! We will merge https://github.com/jwasham/coding-interview-university/pull/78 when we are done.

Have a look to all the work that has already been done, cheers ! 😄

**Comment by YoSaucedo** [comment]
I think is ready for the merge.
If someone can make a review please do it. :)

**Comment by YoSaucedo** [comment]
I think is ready for the merge.
If someone can make a review please do it. :)

File: translations/README_es.md [hunk 5]
```diff
@@ -0,0 +1,1916 @@
...
+
+Es un plan largo, puede tome unos meses; pero si le resulta familiar mucho de esto le tomara mucho menos tiempo.
+
+## Como usarlo
+
+Todo lo que aparece abajo es un plan, y debería abordar los elementos en orden de arriba a abajo.
+
+Estoy usando el tema especial de markdown de Github, incluyendo listas de tareas para comprobar el progreso.
+
+[] **Cree una nueva rama para que pueda validar elementos como este, solo ponga una x en los corchetes: [x]**
```
Comment (dossorio) [inline translations/README_es.md]: We can remove these square brackets

File: translations/README_es.md [hunk 6]
```diff
@@ -0,0 +1,1916 @@
...
+## No sienta no es lo suficientemente listo
+- Los ingenieros de software exitosos son inteligentes pero muchos tienen la inseguridad de no ser lo suficientemente listos.
+- [The myth of the Genius Programmer](https://www.youtube.com/watch?v=0SARbwvhupQ)
+- [It's Dangerous to Go Alone: Battling the Invisible Monsters in Tech](https://www.youtube.com/watch?v=1i8ylq4j_EY)
+
+## Acerca de los recursos de vídeo
+Algunos videos solo están disponibles inscribiéndose a una clase de Coursera, EdX, o Lynda.com. Estos son llamados MOOCs.
+En ocasiones las clases no están en sesión por lo que tendrá que esperar un par de meses, por lo que no tiene acceso. Los cursos de Lynda.com no son gratuitos.
+
+Apreciaría su ayuda añadiendo fuentes siempre disponibles, publicas y gratuitas como videos de Youtube para acompañar a los videos del curso en línea.
```
Comment (dossorio) [inline translations/README_es.md]: This paragraph needs to match the indentation of the next one to be in the same grey box

File: translations/README_es.md [hunk 7]
```diff
@@ -0,0 +1,1916 @@
...
+    - [ ] BFS con matriz de adyacencia
+    - [ ] Ruta de acceso de una sola fuente más corta (Dijkstra)
+    - [ ] Árbol de expansión mínimo
+    - Algoritmos basados en DFS (ver videos de Aduni arriba):
+        - [ ] Comprobar el ciclo (necesario para el tipo topológico, ya que vamos a comprobar el ciclo antes de comenzar)
+        - [ ] Clasificación topológica
+        - [ ] Cuenta los componentes conectados en un grafo
+        - [ ] Lista de componentes fuertemente conectados
+        - [ ] Verificación de grafo bipartito
+Obtendrá más práctica de grafos en el libro de Skiena (vea Sección de libros debajo) y en los libros de entrevistas
```
Comment (dossorio) [inline translations/README_es.md]: We need an extra line break before `Obtendrá`, so it doesn't get mixed with la previous point

File: translations/README_es.md [hunk 8]
```diff
@@ -0,0 +1,1916 @@
...
+        - [ ] Lista de problemas individuales de PD (cada uno es corto):
+            [Dynamic Programming (video)](https://www.youtube.com/playlist?list=PLrmLmBdmIlpsHaNTPP_jHHDx_os9ItYXr)
+    - [ ] Notas de la clase de Yale:
+        - [ ] [Dynamic Programming](http://www.cs.yale.edu/homes/aspnes/classes/223/notes.html#dynamicProgramming)
+    - [ ] Coursera:
+        - [ ] [The RNA secondary structure problem (video)](https://www.coursera.org/learn/algorithmic-thinking-2/lecture/80RrW/the-rna-secondary-structure-problem)
+        - [ ] [A dynamic programming algorithm (video)](https://www.coursera.org/learn/algorithmic-thinking-2/lecture/PSonq/a-dynamic-programming-algorithm)
+        - [ ] [Illustrating the DP algorithm (video)](https://www.coursera.org/learn/algorithmic-thinking-2/lecture/oUEK2/illustrating-the-dp-algorithm)
+        - [ ] [Running time of the DP algorithm (video)](https://www.coursera.org/learn/algorithmic-thinking-2/lecture/nfK2r/running-time-of-the-dp-algorithm)
+        - [ ] [DP vs. recursive implementation (video)](https://www.coursera.org/learn/algorithmic-thinking-2/lecture/M999a/dp-vs-recursive-
```
Comment (dossorio) [inline translations/README_es.md]: We are missing the `Once You've Got The Job` title here

Comment (dossorio) [inline translations/README_es.md]: We need an extra line break here

Comment (dossorio) [inline translations/README_es.md]: We could translate this title to `Recolección de basura` what do you think ?

Comment (dossorio) [inline translations/README_es.md]: curioso

Comment (dossorio) [inline translations/README_es.md]: We are missing the `Una vez obtenido el empleo` title here

Comment (dossorio) [inline translations/README_es.md]: We could translate this title to Recolección de basura what do you think ?

File: translations/README_es.md [hunk 9]
```diff
@@ -0,0 +1,1916 @@
...
+    - [Compiladores](#compiladores)
+    - [Emacs y Vi(m)](#emacs-y-vim)
+    - [Unix: Herramientas para la línea de comandos](#unix-herramientas-para-la-línea-de-comandos)
+    - [Teoría de la información](#teoría-de-la-información)
+    - [Código de paridad y Hamming](#código-de-paridad-y-hamming)
+    - [Entropía](#entropía)
+    - [Criptografía](#criptografía)
+    - [Compresión](#compresión)
+    - [Seguridad informática](#seguridad-informática)
+    - [Garbage collection](#garbage-collection)
```
Comment (dossorio) [inline translations/README_es.md]: We could translate this title to Recolección de basura what do you think ?

File: translations/README_es.md (hunk 8 above)
Comment (dossorio) [inline translations/README_es.md]: We need an extra line break before `- [ ] [Coursera (Scala)]`

Comment (dossorio) [inline translations/README_es.md]: curioso

Comment (dossorio) [inline translations/README_es.md]: We are missing the `Una vez obtenido el empleo` title here

File: README.md [hunk 10]
```diff
@@ -1607,9 +1608,6 @@ software engineer, and to be aware of certain technologies and algorithms, so yo
     - [ ] [Deep Dive Python: Garbage Collection in CPython (video)](https://www.youtube.com/watch?v=P-8Z0-MhdQs&list=PLdzf4Clw0VbOEWOS_sLhT_9zaiQDrS5AR&index=3)
 
 - ### Parallel Programming
-    - [ ] [Coursera (Scala)](https://www.coursera.org/learn/parprog1/home/week/1)
```
Comment (YoSaucedo) [inline README.md]: Only one. Done

File: translations/README_es.md (hunk 5 above)
Comment (YoSaucedo) [inline translations/README_es.md]: I have erased the brackets

File: translations/README_es.md (hunk 6 above)
Comment (YoSaucedo) [inline translations/README_es.md]: Now they have the same indentation.

File: translations/README_es.md (hunk 7 above)
Comment (YoSaucedo) [inline translations/README_es.md]: Space added.

File: translations/README_es.md (hunk 9 above)
Comment (YoSaucedo) [inline translations/README_es.md]: Accepted and Done.

**Comment by dossorio** [comment]
Hi guys ! 

LGTM ! if you are all happy too, I think is ready to merge 👍 

English is key in Software Engineering, but I think bringing this list to our Spanish speaking community is really positive and a starting point for those who want to improve and be better 😃 

Thanks very much to all those who contributed !! and @jwasham for creating it and spreading the word of what a competent Software Engineer should be

Cheers all !

**Comment by jwasham** [comment]
Many thanks to all of you for your contributions, and @dossorio for managing this long-running pull request. I greatly appreciate all your contributions that will help many!

**Pull request merged**